gguf-py/gguf/metadata.py

```diff
@@ -0,0 +1,232 @@
+from __future__ import annotations
+
+import re
+import json
+import frontmatter
+from pathlib import Path
+from typing import Optional
+from dataclasses import dataclass
+from .constants import Keys
+
+
+@dataclass
+class Metadata:
+    # Authorship Metadata to be written to GGUF KV Store
+    name: Optional[str] = None
+    basename: Optional[str] = None
+    finetune: Optional[str] = None
+    author: Optional[str] = None
+    quantized_by: Optional[str] = None
+    organization: Optional[str] = None
+    version: Optional[str] = None
+    base_version: Optional[str] = None
+    url: Optional[str] = None
+    description: Optional[str] = None
+    license: Optional[str] = None
+    license_name: Optional[str] = None
+    license_link: Optional[str] = None
+    source_url: Optional[str] = None
+    source_hf_repo: Optional[str] = None
+    parameter_weight_class: Optional[str] = None
+    tags: Optional[list[str]] = None
+    languages: Optional[list[str]] = None
+    datasets: Optional[list[str]] = None
+
+    @staticmethod
+    def load(metadata_override_path: Optional[Path] = None, model_path: Optional[Path] = None, model_name: Optional[str] = None) -> Metadata:
+        # This grabs as many contextual authorship metadata as possible from the model repository
+        # making any conversion as required to match the gguf kv store metadata format
+        # as well as giving users the ability to override any authorship metadata that may be incorrect
+
+        # Create a new Metadata instance
+        metadata = Metadata()
+
+        # load huggingface model card if available
+        # Reference Model Card Metadata: https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1
+        model_card = Metadata.load_model_card(model_path)
+
+        if "model_name" in model_card:
+            # Not part of huggingface model card standard but notice some model creator using it
+            # such as TheBloke who would encode 'Mixtral 8X7B Instruct v0.1' into model_name
+            metadata.name = model_card.get("model_name")
+
+        if "base_model" in model_card:
+            # Not part of huggingface model card standard but notice some model creator using it
```

Comment:
it is actually in the "spec": https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1 
and documented here: https://huggingface.co/docs/hub/en/model-cards#specifying-a-base-model